ui/components/app/modals/modal.js

```diff
@@ -111,7 +112,38 @@ const MODALS = {
       height: '100%',
     },
   },
-
+  DEPOSIT_TOKEN: {
```

Comment:
I think this is the old approach of how we trigger modals by dispatching an action, the new approach is more simplified by relying on local state change to trigger a conditional rendering of the modal component, a clear example of the modern approach can be seen in the following file `/metamask-extension/ui/components/ui/sender-to-recipient/sender-to-recipient.component.js` 
`      {showNicknamePopovers ? (
        <NicknamePopovers
          onClose={() => setShowNicknamePopovers(false)}
          address={checksummedRecipientAddress}
        />
      ) : null}`

Reply:
agreed to keep the "old approach" for triggering modals. while it can be an overkill for a modal, dispatching this new modal, respects the same strategy used to show the ether modal. choices were:

- refactor ether modal (increased too much the scope)
- keep ether modal as it is, and use the local state approach for the tokens modal (confusing to have two strategies for pretty much the same modals)
- use the dispatch approach for both (selected)

Reply:
for consistency purposes, it would make sense to keep the same strategy, but moving forward, we need to deviate from using the dispatch approach when mounting/un-mounting modals

Reply:
resolved

Reply:
We need to convert to using the new popover component and not extending our deprecated code patterns, so i am unresolving this conversation and marking it as a requirement to move forward.

Reply:
Can you say more as to why you have to refactor the old modal to implement a new one in the appropriate way?

Reply:
commented here https://github.com/MetaMask/metamask-extension/pull/15551#discussion_r964962200

btw i updated screenshots to show how it looks now that its using the design system components